test/app.test.js

```diff
@@ -1,10 +1,250 @@
-import { getNearestShops } from '../src/app';
-
-describe('App', () => {
-  it('should return an array when the input is valid', () => {
-    expect(Array.isArray(getNearestShops({
-      lat: 0,
-      lng: 0,
-    }))).toBe(true);
+import { getNearestShops } from "../src/app";
+import DeltaCoffeeShop from "../src/deltaCoffeeShop";
+import DeltaCoffeeShopList from "../src/deltaCoffeeShopList";
+import SortedDeltaCoffeShopList from "../src/sortedDeltaCoffeeShopList";
+import errorCodes from "../src/errorcodes.js";
+import { ERROR_TOKEN, errorResponse } from "../src/utils.js";
+
+describe("App", () => {
```

Comment:
**What I like**: you added tests, quite a few of them, seems very diligent.
**What could be better**: splitting tests for different modules/classes into different files so it is easier to focus on one thing at a time and not scroll so much. It is great when you open the file and you can see at a glance what is happening. I also don't like some of the tests because they seem to do nothing, like the ones where you manually create the errors in the test and then expect them to be what you just created, it proves nothing.

Reply:
One cool thing about writing tests is that it immediately shows if the requirements for the feature are well understood. I can just collapse everything and read just the titles of these tests and I can get an idea of how your program is supposed to work. It also shows if your program is built to do something wrong. Once again, love that you wrote tests :+1: